internal/usbgadget/hid_keyboard.go

```diff
@@ -211,22 +261,112 @@ func (u *UsbGadget) keyboardWriteHidFile(data []byte) error {
 	return nil
 }
 
-func (u *UsbGadget) KeyboardReport(modifier uint8, keys []uint8) error {
+func (u *UsbGadget) KeyboardReport(modifier byte, keys []byte) error {
 	u.keyboardLock.Lock()
 	defer u.keyboardLock.Unlock()
+	defer u.resetUserInputTime()
 
-	if len(keys) > 6 {
-		keys = keys[:6]
+	if len(keys) > hidKeyBufferSize {
+		keys = keys[:hidKeyBufferSize]
 	}
-	if len(keys) < 6 {
-		keys = append(keys, make([]uint8, 6-len(keys))...)
+	if len(keys) < hidKeyBufferSize {
+		keys = append(keys, make([]byte, hidKeyBufferSize-len(keys))...)
 	}
 
-	err := u.keyboardWriteHidFile([]byte{modifier, 0, keys[0], keys[1], keys[2], keys[3], keys[4], keys[5]})
-	if err != nil {
-		return err
+	return u.keyboardWriteHidFile(modifier, keys)
+}
+
+const (
```

Comment:
the device is now responsible for tracking the modifier state, so it needs the keycodes for those (from the HID spec)